flags.go

```diff
@@ -74,19 +76,39 @@ func BuildFlags(obj any) Flags {
 	if v.Kind() != reflect.Struct {
 		panic(fmt.Errorf("expected a struct, got %s", v.Kind()))
 	}
+
+	return buildFlagsRecursive(v)
+}
+
+func buildFlagsRecursive(v reflect.Value) Flags {
 	t := v.Type()
+	var flags Flags
 
-	var err error
-	flags := make([]Flag, t.NumField())
 	for i := 0; i < t.NumField(); i++ {
-		flags[i], err = buildFlag(v.Field(i), t.Field(i))
-		if err != nil {
-			panic(err)
+		field := t.Field(i)
+		fieldValue := v.Field(i)
+
+		// Only process fields with a 'short' or 'long' tag
+		if hasTag(field.Tag, "short") || hasTag(field.Tag, "long") {
+			flag, err := buildFlag(fieldValue, field)
+			if err != nil {
+				panic(err)
+			}
+			flags = append(flags, flag)
+		} else if fieldValue.Kind() == reflect.Struct {
```

Comment:
@hariso that could be an option. I would need to check how to do that accordingly. Would it make sense maybe ship this PR and add that separately to maybe focus on shipping https://github.com/ConduitIO/conduit/pull/1983? It's turning out a bit more complicated than I had planned and working on this would delay the other one.

Reply:
Makes sense to me!

Reply:
@hariso thank you for creating that other issue.